homeassistant/components/calendar/trigger.py

```diff
@@ -87,18 +90,17 @@ async def _fetch_events(self, now: datetime.datetime) -> None:
         # returned events may have already started but matched the start/end time
         # filtering above, so exclude any events that have already passed the
         # trigger time.
-        event_list = [
-            (dt_util.as_utc(event.start_datetime_local), event) for event in events
-        ]
+        event_list = []
+        for event in events:
+            event_time = (
+                event.start_datetime_local
+                if self._event_type == EVENT_START
+                else event.end_datetime_local
```

Comment:
Will `async_get_events` give us events that _start_ after start date or any event that starts prior to `start_date` but end before `end_date` ?

Reply:
The latter. The spec is here: https://developers.home-assistant.io/docs/core/entity/calendar#get-events
You can think of it like the start is applied to the end and the end is applied to the start, so you get back any event active in the window.


The behavior was previously not spelled out previously, so i think i should do another audit of the core implementations to ensure they conform.